language/tools/move-package/tests/test_runner.rs

```diff
@@ -2,135 +2,164 @@
     ffi::OsStr,
     fs,
-    path::{Component, Path, PathBuf},
+    path::{Path, PathBuf},
 };
-use tempfile::tempdir;
+use tempfile::{tempdir, TempDir};
 
-const COMPILE_EXT: &str = "compile";
-const MODEL_EXT: &str = "model";
+const EXTENSIONS: &[&str] = &["resolved", "compiled", "modeled"];
 
 pub fn run_test(path: &Path) -> datatest_stable::Result<()> {
     package_hooks::register_package_hooks(Box::new(TestHooks()));
     let update_baseline = read_env_update_baseline();
-    if path
-        .components()
-        .any(|component| component == Component::Normal(OsStr::new("deps_only")))
-    {
+
+    if path.iter().any(|part| part == "deps_only") {
+        return Ok(());
+    }
+
+    let test = Test::from_path(path)?;
+    let output = test.run().unwrap_or_else(|err| format!("{:#}\n", err));
+
+    if update_baseline {
+        fs::write(&test.expected, &output)?;
         return Ok(());
     }
-    let exp_path = path.with_extension(EXP_EXT);
-    let should_compile = path.with_extension(COMPILE_EXT).is_file();
-    let should_model = path.with_extension(MODEL_EXT).is_file();
-
-    let exp_exists = exp_path.is_file();
-
-    let contents = fs::read_to_string(path)?;
-    let output = match MP::parse_move_manifest_string(contents)
-        .and_then(MP::parse_source_manifest)
-        .and_then(|parsed_manifest| {
-            RG::ResolutionGraph::new(
-                parsed_manifest,
-                path.parent().unwrap().to_path_buf(),
-                BuildConfig {
-                    dev_mode: true,
-                    test_mode: false,
-                    generate_docs: false,
-                    generate_abis: false,
-                    install_dir: Some(tempdir().unwrap().path().to_path_buf()),
-                    force_recompilation: false,
-                    ..Default::default()
-                },
-                &mut Vec::new(), /* empty writer as no diags needed */
+
+    let expected = fs::read_to_string(&test.expected)?;
+    if expected != output {
+        return Err(anyhow!(add_update_baseline_fix(format!(
+            "Expected outputs differ for {:?}:\n{}",
+            test.expected,
+            format_diff(expected, output),
+        )))
+        .into());
+    }
+
+    Ok(())
+}
+
+struct Test<'a> {
+    toml_path: &'a Path,
+    expected: PathBuf,
+    output_dir: TempDir,
+}
+
+impl Test<'_> {
+    fn from_path(toml_path: &Path) -> datatest_stable::Result<Test> {
+        let mut candidates = EXTENSIONS
+            .iter()
+            .map(|ext| toml_path.with_extension(*ext))
+            .filter(|p| p.is_file());
+
+        let Some(expected) = candidates.next() else {
+            return Err(anyhow!(
+                "No snapshot file found for {:?}, please add a file with the same basename and one \
+                 of the following extensions: {:#?}\n\n\
+                 You probably want to re-run with `env UPDATE_BASELINE=1` after adding this file.",
+                toml_path,
```

Comment:
So is my understanding correct that, with this change, running the test with UPDATE_BASELINE=1 set will not automatically generate the snapshot files if they do not exist?

Reply:
That's correct, this becomes more palatable after #759, when a given test can have multiple snapshots.